### ISSUE TITLE
컨테이너 스케줄러 관련 기존 기술 요약 및 정리

### ISSUE THREAD

**Issue by hyu-leeky** [ddps-lab/edge-management-module#15]
엣지 기반 K8S 스케줄러 논문 작성 시 필요한 관련 기술에 대한 세부 사항 정리

**Comment by hyu-leeky** [comment]
KCSS: Kubernetes container scheduling strategy
https://link.springer.com/article/10.1007/s11227-020-03427-3

위 논문이 간단한 알고리즘을 시작으로 관련한 기술에 대한 소개가 있음. 해당 논문을 시작으로 관련된 논문을 다 읽어보고 자세하게 정리해보자.
위 논문을 논문 세미나에서 진행해도 좋을듯.

**Comment by jungae-park** [comment]
네 교수님 안그래도 저 논문을 다운로드 받으려고 들어가봤는데, 저는 다운로드 받을 수 있는 접근 권한이 없는것 같습니다. (저는 구매해야하는 것 같습니다.)
혹시 문서로 공유해주실 수 있을까요?

**Comment by hyu-leeky** [comment]
학교에서는 다운로드 가능하지 않아?

**Comment by jungae-park** [comment]
제가 지금 학교임에도 불구하고 다운로드가 되지 않습니다.

**Comment by hyu-leeky** [comment]
그래? 접속되는 학교 망의 종류가 다른가.. 논문은 첨부함.
[k8s-scheduling-supercomputing-2021.pdf](https://github.com/ddps-lab/edge-management-module/files/9630106/k8s-scheduling-supercomputing-2021.pdf)

**Comment by jungae-park** [comment]
네 교수님 감사합니다.
요즘 학교 네트워크 망이 불안정하고 이상한 경우가 있기는 합니다.
(저는 KMU5G를 사용하여 논문을 다운로드 하려고 하였었습니다.)
해당 논문을 꼼꼼히 읽어보고 자세하게 정리해 보겠습니다.

**Comment by jungae-park** [comment]
공유해주신 KCSS: Kubernetes container scheduling strategy 논문을 자세히 읽고 아래와 같이 요약해보았습니다.
자세하게는 노션으로 정리도 해두었습니다.
https://encouraging-lantern-5d1.notion.site/KCSS-Kubernetes-container-scheduling-strategy-09bf44ac748d4e88abb49f902b3fb9c9

1. kubernetes 컨테이너 스케줄링 전략을 위해 새로운 KCSS 알고리즘을 소개
2. 사용자의 요구사항에 따라 컨테이너 일정을 최적화하는 것을 목표로하며 6가지 주요 기준을 고려함
    1. 각 노드의 CPU 사용률
    2. 각 노드의 메모리 활용률
    3. 각 디스크 활용률
    4. 각 노드의 전력 소비
    5. 각 노드에서 실행 중인 컨테이너 수
    6. 사용자가 컨테이너에 대해 선택한 이미지를 전송하는 시간
3.  KCSS는 단일 순위에서 모든 기준을 집계하는 다중 기준 결정 분석 알고리즘 기반으로 TOPSIS(Similarity to ldeal Solution) 알고리즘에 의한 우선순위 지정 기법
4. 가장 높은 우선 순위를 가진 노드를 선택하여 컨테이너를 실행

말씀해주셨듯이 논문세미나에서 돌아오는 차례에 발표를 진행해보겠습니다.

그리고 이외에 다른 관련 논문들도 정리해서 이 이슈에 기록해 두겠습니다.

**Comment by hyu-leeky** [comment]
응 그러자. 잘했다. TOPSIS 알고리즘이 간단해보이면서도 꽤나 유용해 보이는듯 하다. 관련 기술을 잘 정리해보자.

**Comment by jungae-park** [comment]
- Building Heterogeneous Cloud System for Machine Learning Inference
    
![image](https://user-images.githubusercontent.com/46885199/198341484-b31b40ac-03b1-4972-9185-f8bc3cf5754d.png)

    
    - QoS를 충족하며 쿼리 처리량을 최대화하는 Kairos 프레임워크 제안
    - Jonker-Volgenant 알고리즘 사용
    - 헝가리 알고리즘의 발전된 알고리즘이고, 속도가 더 빠름
    - 헝가리 알고리즘은 예를 들어 세 개의 작업자가 있을 때, 각각 다른 작업을 수행하고 작업 시간을 측정할때 세 개의 작업자가 작업을 수행할 때 적은 시간으로 작업을 수행하도록 하기 위한 알고리즘
    
![image](https://user-images.githubusercontent.com/46885199/198341540-7584ee03-5249-48ea-8053-01205e70c786.png)

**Comment by jungae-park** [comment]
위의 Building Heterogeneous Cloud System for Machine Learning Inference 논문에서 실험에 사용된 워크로드에 대한 내용이 있어 기록합니다.

![image](https://user-images.githubusercontent.com/46885199/203479759-f4167ff7-dd4f-4395-8cf5-49d1072ffbcb.png)

facebook에서 실제 쿼리에 따른 작업을 활용하였습니다.
어떤식으로 사용되었는지 확인해보고 있습니다.

**Comment by jungae-park** [comment]
EmBench: Quantifying Performance Variations of Deep Neural Networks across Modern Commodity Devices 논문에서는 다양한 상용 컴퓨팅 플랫폼을 대상으로 벤치마킹하여 수집한 데이터를 분석하기도 하였습니다.

![image](https://user-images.githubusercontent.com/46885199/203480752-a14dac61-2ec0-47f0-9d06-26eebb51f826.png)

![image](https://user-images.githubusercontent.com/46885199/203480915-4982e958-64ba-43cd-a452-f389c7915a3b.png)

**Comment by hyu-leeky** [comment]
응 정애야. 저건 그냥 모델일것 같기는 한데. 해당 모델을 활용해서 어떻게 사용자의 요청이 들어오는지를 (빈도, 도착시간 등) 서술한 자료나 로그등이 있다면 참조하면 좋을것 같아.
